addons/account/models/account_move.py

```diff
@@ -3320,7 +3324,7 @@ def action_invoice_print(self):
         """ Print the invoice and mark it as sent, so that we can see more
             easily the next step of the workflow
         """
-        if any(not move.is_invoice(include_receipts=True) for move in self):
+        if any(not move._can_be_emailed() for move in self):
```

Comment:
You shouldn't allow people to use this action for something else than invoices. If your purpose is to allow the user to download an additional document, you just need to allow it in some way without hacking the business flow.